src/data_conn.c

```diff
@@ -600,13 +603,21 @@ replica_thread(void *arg)
 			if (ret == -1)
 				goto exit;
 		}
-		/* Need to handle wait IOs that are taking more than IO timeout */
-		// handle_data_conn_error(r);
-		// return -1;
+
+		/* check for replica IO timeout */
+		pending_cmd = TAILQ_FIRST(&r->waitq);
+		if (pending_cmd != NULL) {
+			clock_gettime(CLOCK_MONOTONIC, &now);
+			timesdiff(pending_cmd->queued_time, now, diff);
```

Comment:
let us change epoll_timeout based on when the next iteration should be, so that, an IO doesn't hang for time longer than the configured replica timeout

Reply:
ok